test/tests/wa_has_one_optimization_test.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+require_relative "../test_helper"
+
+# has_one requires doing sub-queries with SELECT IN / applying limits to
+# handle conditions / counts properly. However, when there is no conditions and counts,
+# the deepest layer can avoid those sub-queries, which is faster.
+
+module RegexpTricks
```

Comment:
If the `squeeze` approach works you can also remove this module - but I suppose it would be good to have a code comment somewhere to say why `squeeze` is used